mitm_test.go

```diff
@@ -393,6 +393,49 @@ func TestWebsocket(t *testing.T) {
 	testEchoServer(wcServer.URL, proxyConn, t)
 }
 
+func skipAll(req *http.Request) bool {
+	return true
+}
+
+func TestSkipRequest(t *testing.T) {
+	p := &Proxy{
+		Wrap:     shouldNotReach(t),
+		Director: HTTPDirector,
+		SkipRequest: skipAll,
+	}
+	l, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatal("Listen:", err)
+	}
+	defer l.Close()
+
+	go func() {
+		if err := http.Serve(l, p); err != nil {
+			if !strings.Contains(err.Error(), "use of closed network") {
+				t.Fatal("Serve:", err)
+			}
+		}
+	}()
+
+	req, _ := http.NewRequest("GET", "http://www.google.com/", nil)
```

Comment:
I think this is actually calling out to google? Let's try to keep the unit tests independent. Can you set up a test server (with httptest.NewServer) instead?